### ISSUE TITLE
extract: make patterns relative

### ISSUE THREAD

**Issue by ThomasWaldmann** [borgbackup/borg#2763]
From a (rejected, non-subscriber) mailing list post:

> 2/ the need for relative paths when extracting files that were stored
> with absolute paths. 
> 
> When one needs to extract something, one is usually a bit stressed, and
> it's utterly frustrating to get the message "Include pattern
> '/etc/group' never matched." 
> 
> To check with str.startswith('/') (and hint to the user that the path
> has to be relative), might be a quick fix: "Did you intend to extract
> 'etc/group' instead?"

**Comment by ThomasWaldmann** [comment]
I don't like it asking interactively.

But as at create time we automatically store relative pathes into item metadata (stripping the leading "/"), we could also strip leading "/" at extract time.

**Comment by enkore** [comment]
The usual disclaimer: Only works for fm/sh/pp/pf, not re.

**Comment by ThomasWaldmann** [comment]
see also #2644.

**Comment by rugk** [comment]
"see also" is good, is not this actually a duplicate of https://github.com/borgbackup/borg/issues/2644?

**Comment by enkore** [comment]
PATH vs pattern

**Comment by enkore** [comment]
True, "PATH" in borg list are actually patterns as well.

**Comment by rugk** [comment]
Okay, usually one marks the newer issue as a duplicate of the older one and not vice versa, but I don't care for this soo much… :smiley:

**Comment by fantasya-pbem** [comment]
Isn't this issue solved now, after #4029 has been solved?

**Comment by ThomasWaldmann** [comment]
Yup, closing.
